src/azure-cli-core/azure/cli/core/commands/parameters.py

```diff
@@ -60,19 +60,32 @@ def completer(prefix, action, parsed_args, **kwargs):#pylint: disable=unused-arg
             return [r.name for r in get_resources_in_subscription(resource_type=resource_type)]
     return completer
 
-def get_enum_type_completion_list(enum_type=None):
-    ENUM_MAPPING_KEY = '_value2member_map_'
-    def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-argument
-        return list(enum_type.__dict__[ENUM_MAPPING_KEY].keys())
-    return completer
-
 def get_generic_completion_list(generic_list):
     def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-argument
         return generic_list
     return completer
 
-def get_enum_choices(enum_type):
-    return [x.value for x in enum_type]
+class CaseInsenstiveList(list): # pylint: disable=too-few-public-methods
```

Comment:
I deliberately preserve the enum values' original casing for a few reason:

1. camelCaseIsOftenMoreReadable than camelcaseisoftenmorereadable when it appears in help.
2. preserving the case allows me to use this with services that are case sensitive. If I altered the casing of the choices, I couldn't do this or would need a wonky map back from lowercase values to properly cased ones
3. Some instances in the code forced everything to uppercase and that is equally valid

The comparisons are all done in lowercase.